examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,335 @@
+    "target_points", target_points_num);
+  auto source_points_host = Kokkos::create_mirror_view(source_points);
+  auto target_points_host = Kokkos::create_mirror_view(target_points);
+
+  // Generate source points
+  source_points_host(0) = ArborX::Point {  1.,  1.,  0. };
+  source_points_host(1) = ArborX::Point { -1.,  1.,  0. };
+  source_points_host(2) = ArborX::Point { -1., -1.,  0. };
+  source_points_host(3) = ArborX::Point {  1., -1.,  0. };
+  source_points_host(4) = ArborX::Point {  0.,  0.,  1. };
+  source_points_host(5) = ArborX::Point {  1.,  1.,  2. };
+  source_points_host(6) = ArborX::Point { -1.,  1.,  2. };
+  source_points_host(7) = ArborX::Point { -1., -1.,  2. };
+  source_points_host(8) = ArborX::Point {  1., -1.,  2. };
+  source_points_host(9) = ArborX::Point {  0.,  0., -1. };
+  Kokkos::deep_copy(source_points, source_points_host);
+
+  // Generate target points
+  target_points_host(0) = ArborX::Point {  0.,  0.,  0. };
+  target_points_host(1) = ArborX::Point {  .5,  .5,  0. };
+  target_points_host(2) = ArborX::Point { -.5,  .5,  1. };
+  target_points_host(3) = ArborX::Point {  0., -.5, 1.5 };
+  Kokkos::deep_copy(target_points, target_points_host);
+
+  // Arrange source points as tree
+  auto source_tree =
+    ArborX::BVH<MemorySpace>(ExecutionSpace{}, source_points);
+  
+  // Create the queries
+  // For each target point we query the closest source points
+  auto queries = Kokkos::View<ArborX::Nearest<ArborX::Point>*, MemorySpace>(
+    "queries", target_points_num);
+  Kokkos::parallel_for(
+    "make_queries",
+    Kokkos::RangePolicy<ExecutionSpace>(0, target_points_num),
+    KOKKOS_LAMBDA (const int i) {
+      queries(i) = ArborX::nearest(target_points(i), num_neighbors);
+  });
+
+  // Perform the query
+  auto indices = Kokkos::View<int *, MemorySpace>("indices", 0);
+  auto offsets = Kokkos::View<int *, MemorySpace>("offsets", 0);
+  source_tree.query(ExecutionSpace{}, queries, indices, offsets);
+
+  // Now that we have the neighbors, we recompute their position using
+  // their target point as the origin.
+  // This is used as an optimisation later in the algorithm
+  auto tr_source_points = Kokkos::View<ArborX::Point**, MemorySpace>(
+    "tr_source_points", target_points_num, num_neighbors);
+  Kokkos::parallel_for(
+    "transform_source_points",
+    Kokkos::RangePolicy<ExecutionSpace>(0, target_points_num),
+    KOKKOS_LAMBDA (const int i) {
+      for (int j = offsets(i); j < offsets(i+1); j++) {
+        tr_source_points(i, j - offsets(i)) = ArborX::Point {
+          source_points(indices(j))[0] - target_points(i)[0],
+          source_points(indices(j))[1] - target_points(i)[1],
+          source_points(indices(j))[2] - target_points(i)[2],
+        };
+      }
+  });
+
+  // Compute the radii for the weight (phi) vector
+  auto radii = Kokkos::View<double*, MemorySpace>(
+    "radii", target_points_num);
+  constexpr double epsilon = std::numeric_limits<double>::epsilon();
```

Comment:
Probably want `float` here, it's up for a discussion.

Reply:
It depends a lot on how the values should be computed. Should the coefficients be handled as doubles or floats?

Reply:
It may make sense to do the calculations based on the floating type of the data. We have `Point<DIM, FloatingPoint>`, so should probably use the same type as the second template argument of the `Point`. Note that we need #918 with `GeometricTraits::coordinate_type<Point>::type` to figure out that type.